integration/http_test.go

```diff
@@ -0,0 +1,273 @@
+			json:   responseJSON.GetPath("result", "first_name"),
+			expect: `"Wilfred"`,
+		},
+		{
+			name:   "contact middle name",
+			json:   responseJSON.GetPath("result", "middle_name"),
+			expect: `"Wallace"`,
+		},
+		{
+			name:   "contact last",
+			json:   responseJSON.GetPath("result", "last_name"),
+			expect: `"O'Docker"`,
+		},
+		{
+			name:   "contact notes",
+			json:   responseJSON.GetPath("result", "notes"),
+			expect: `"build the containers at dusk"`,
+		},
+		{
+			name:   "contact primary email",
+			json:   responseJSON.GetPath("result", "primary_email"),
+			expect: `"test@test.com"`,
+		},
+		{
+			name:   "contact email list",
+			json:   responseJSON.GetPath("result", "emails"),
+			expect: `[{"address":"test@test.com","id":"1"}]`,
+		},
+		{
+			name:   "success response",
+			json:   responseJSON.GetPath("success"),
+			expect: `{"code":"OK","status":200}`,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			ValidateJSONSchema(t, test.json, test.expect)
+		})
+	}
+}
+
+// TestInvalidEmail_REST attempts to create a new contact that has an invalid
+// email address.
+// 1. Use the REST API to create a new contact with an invalid email
+// 2. Ensure the HTTP response status code is 400
+// 3. Ensure the HTTP response has a detailed error message
+func TestInvalidEmail_REST(t *testing.T) {
+	dbTest.Reset(t)
+	contact := pb.Contact{
+		PrimaryEmail: "invalid-email-address",
+	}
+	res, err := MakeRequestWithDefaults(
+		http.MethodPost, "http://localhost:8080/v1/contacts",
+		contact,
+	)
+	if err != nil {
+		t.Fatalf("unable to create contact %v", err)
+	}
+	ValidateResponseCode(t, res, http.StatusBadRequest)
+	responseJSON, err := simplejson.NewFromReader(res.Body)
+	if err != nil {
+		t.Fatalf("unable to unmarshal response json: %v", err)
+	}
+	var tests = []struct {
+		name   string
+		json   *simplejson.Json
+		expect string
+	}{
+		{
+			name:   "check response code",
+			json:   responseJSON.GetPath("error", "code"),
+			expect: `"INVALID_ARGUMENT"`,
+		},
+		{
+			name:   "check http status",
+			json:   responseJSON.GetPath("error", "status"),
+			expect: `400`,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			ValidateJSONSchema(t, test.json, test.expect)
+		})
+	}
+}
+
+// TestDeleteContact_REST uses the REST gateway to create a new contact and
+// ensure it can get deleted correctly
+// 1. Create a contact entry with a POST request
+// 2. Delete the contact
+// 3. Ensure the DELETE response matches schema is expected
+func TestDeleteContact_REST(t *testing.T) {
+	dbTest.Reset(t)
+	contact := pb.Contact{
+		PrimaryEmail: "test@test.com",
+	}
+	_, err := MakeRequestWithDefaults(
+		http.MethodPost,
+		"http://localhost:8080/v1/contacts",
+		contact,
+	)
+	if err != nil {
+		t.Fatalf("unable to create contact: %v", err)
+	}
+	res, err := MakeRequestWithDefaults(
+		http.MethodDelete,
+		"http://localhost:8080/v1/contacts/1",
```

Comment:
it would be safer to read the ID from the `POST` response and use it here, rather than hard-coding `1`

Reply:
Sounds good!